### PR TITLE
[Refactor] Update tests for new core data stack

### PR DESCRIPTION
**Pull request by typfel**
## What's new in this PR?

After the introduction of a new core data stack in https://github.com/wireapp/wire-ios-data-model/pull/1151 we need to update all tests which interact with core data.

## Dependencies

Needs releases with:

- [x] https://github.com/wireapp/wire-ios-data-model/pull/1151

### CODE REVIEW

File: Sources/Object Syncs/Helpers/ZMChangeTrackerBootstrapTests.m [hunk 1]
```diff
@@ -52,7 +53,8 @@ - (void)addTrackedObjects:(NSSet *)objects
 @interface ZMChangeTrackerBootstrapTests : ZMTBaseTest
 
 @property (nonatomic) ZMChangeTrackerBootstrap *sut;
-@property (nonatomic) ZMTestSession *testSession;
+@property (nonatomic) CoreDataStack *coreDataStack;
+//@property (nonatomic) ZMTestSession *testSession;
```
Comment (johnxnguyen) [inline Sources/Object Syncs/Helpers/ZMChangeTrackerBootstrapTests.m:57]: To delete?

File: Sources/Object Syncs/Helpers/ZMChangeTrackerBootstrapTests.m [hunk 2]
```diff
@@ -107,8 +109,9 @@ - (void)tearDown {
     self.changeTracker2 = nil;
     self.connection1 = nil;
     self.conversation2 = nil;
-    [self.testSession tearDown];
-    self.testSession = nil;
+    self.coreDataStack = nil;
+//    [self.testSession tearDown];
+//    self.testSession = nil;
```
Comment (johnxnguyen) [inline Sources/Object Syncs/Helpers/ZMChangeTrackerBootstrapTests.m:114]: To delete?

File: Sources/Object Syncs/Upstream/Helpers/ZMLocallyModifiedObjectSyncStatusTests.m [hunk 3]
```diff
@@ -27,7 +29,8 @@
 
 @interface ZMLocallyModifiedObjectSyncStatusTests : ZMTBaseTest
 
-@property (nonatomic) ZMTestSession *testSession;
+//@property (nonatomic) ZMTestSession *testSession;
```
Comment (johnxnguyen) [inline Sources/Object Syncs/Upstream/Helpers/ZMLocallyModifiedObjectSyncStatusTests.m:32]: To delete?

File: Sources/Request Syncs/ZMTimedSingleRequestSyncTests.m [hunk 4]
```diff
@@ -22,12 +22,14 @@
...
 @import WireRequestStrategy;
 
+#import "WireRequestStrategyTests-Swift.h"
 #import "ZMSingleRequestSync.h"
 #import "ZMTimedSingleRequestSync.h"
 
 @interface ZMTimedSingleRequestSyncTests : ZMTBaseTest <ZMSingleRequestTranscoder>
 
-@property (nonatomic) ZMTestSession *testSession;
+//@property (nonatomic) ZMTestSession *testSession;
```
Comment (johnxnguyen) [inline Sources/Request Syncs/ZMTimedSingleRequestSyncTests.m:31]: To delete?

File: Sources/Object Syncs/Helpers/ZMChangeTrackerBootstrapTests.m [hunk 5]
```diff
@@ -72,28 +74,28 @@ @implementation ZMChangeTrackerBootstrapTests
 - (void)setUp
 {
     [super setUp];
+
+    self.coreDataStack = self.coreDataStack = [self createCoreDataStackWithUserIdentifier:[NSUUID UUID]
```
Comment (KaterinaWire) [inline Sources/Object Syncs/Helpers/ZMChangeTrackerBootstrapTests.m:78]: `self.coreDataStack` is twice

File: Sources/Object Syncs/Downstream/ZMDownstreamObjectSyncOrderingTests.m [hunk 6]
```diff
@@ -40,31 +41,30 @@ @implementation ZMDownstreamObjectSyncOrderingTests
 - (void)setUp
 {
     [super setUp];
-    
-    self.testSession = [[ZMTestSession alloc] initWithDispatchGroup:self.dispatchGroup];
-    [self.testSession prepareForTestNamed:self.name];
 
-    self.conversation1 = [ZMConversation insertNewObjectInManagedObjectContext:self.testSession.uiMOC];
+    self.coreDataStack = self.coreDataStack = [self createCoreDataStackWithUserIdentifier:[NSUUID UUID]
```
Comment (KaterinaWire) [inline Sources/Object Syncs/Downstream/ZMDownstreamObjectSyncOrderingTests.m:45]: `self.coreDataStack` is twice

**Review by johnxnguyen — APPROVED** [review]

**Review by KaterinaWire — APPROVED** [review]

**Pull request merged**